### PR TITLE
Add binder badge to readme

### PR DESCRIPTION
**Pull request by ColCarroll**
This implements #2757 – it looks like at least dask had trouble keeping the examples working, but I think this is worth a try!

### CODE REVIEW

**Comment by junpenglao** [comment]
Binder should work. I implemented for pymc-devs/resources https://github.com/pymc-devs/resources/tree/master/BCM and they are working fine.

**Comment by ColCarroll** [comment]
rst now seems to be rendering properly...

**Comment by junpenglao** [comment]
I think maybe it is better to put it in a separate branch, for one many notebooks requires matplotlib.

**Comment by ColCarroll** [comment]
Good point, matplotlib is not being installed! I'll mark this a WIP until I figure that out – it looks like there are ways to add "extra requirements".  Super close to "just working"!

**Comment by ColCarroll** [comment]
Should be ready to go – see https://github.com/jupyterhub/binder/issues/51 for how.  I ended up adding a few more requirements to `requirements-dev.txt`, so that the notebooks could run. `seaborn`, `ipython`, and `keras`.  As a "test", I executed the "getting started" notebook on my branch, and it worked fine.

Feel free to poke around here (they allow branch builds on forks!), and let me know if there are more notebook requirements that are needed:
https://mybinder.org/v2/gh/colcarroll/pymc3/binder?filepath=%2Fdocs%2Fsource%2Fnotebooks

**Comment by junpenglao** [comment]
I think some notebooks also use bokeh

**Comment by twiecki** [comment]
Just tested it out, this is excellent!

File: README.rst [hunk 1]
```diff
@@ -12,6 +12,10 @@ large suite of problems.
 
 Check out the `getting started guide <http://docs.pymc.io/notebooks/getting_started>`__!
 
+You can also interact with live examples using `Binder <http://mybinder.org/>`__: 
+
+|Binder|
```
Comment (twiecki) [inline README.rst]: Can you add that badge to the top with the other badges?

Comment (ColCarroll) [reply README.rst]: Do you think it is clear enough what it does?  Maybe badge up top and sentence with inline link here? "You can [also interact with live examples](https://mybinder.org/v2/gh/colcarroll/pymc3/binder?filepath=%2Fdocs%2Fsource%2Fnotebooks) using Binder."

Comment (twiecki) [reply README.rst]: Yeah I think that'd be perfect.

**Comment by twiecki** [comment]
Thanks @ColCarroll!

**Pull request merged**